ccl/workloadccl/allccl: TestAllRegisteredSetup failed

ccl/workloadccl/allccl.TestAllRegisteredSetup [failed](https://tanzanite.cluster.engflow.com/invocations/default/2c447162-bee7-4e1f-bd1c-1ed11006d63e?testReportRun=1&testReportShard=2&testReportAttempt=1#targets-Ly9wa2cvY2NsL3dvcmtsb2FkY2NsL2FsbGNjbDphbGxjY2xfdGVzdA==) on master @ [a0f39e7ac9574756063bc90bba6bc532b45c33d4](https://github.com/cockroachdb/cockroach/commits/a0f39e7ac9574756063bc90bba6bc532b45c33d4):

Fatal error:

```
panic: test timed out after 24m57s
	running tests:
		TestAllRegisteredSetup (24m57s)
		TestAllRegisteredSetup/tpccmultidb (20m45s)

```
Stack: 

```
goroutine 102041692 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2373 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:215 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestAllRegisteredSetup_json1588448018
    test_log_scope.go:76: use -show-logs to present logs inline
    all_test.go:173: -- test log scope end --
=== RUN   TestAllRegisteredSetup/ledger
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestAllRegisteredSetup_ledger1419359716
    test_log_scope.go:76: use -show-logs to present logs inline
    all_test.go:173: -- test log scope end --
=== RUN   TestAllRegisteredSetup/movr
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestAllRegisteredSetup_movr1126251896
    test_log_scope.go:76: use -show-logs to present logs inline
    all_test.go:173: -- test log scope end --
=== RUN   TestAllRegisteredSetup/roachmart
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestAllRegisteredSetup_roachmart2718800360
    test_log_scope.go:76: use -show-logs to present logs inline
    all_test.go:173: -- test log scope end --
=== RUN   TestAllRegisteredSetup/startrek
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestAllRegisteredSetup_startrek2450885376
    test_log_scope.go:76: use -show-logs to present logs inline
    all_test.go:173: -- test log scope end --
=== RUN   TestAllRegisteredSetup/connectionlatency
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestAllRegisteredSetup_connectionlatency3354638760
    test_log_scope.go:76: use -show-logs to present logs inline
    all_test.go:173: -- test log scope end --
=== RUN   TestAllRegisteredSetup/geospatial
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestAllRegisteredSetup_geospatial453690263
    test_log_scope.go:76: use -show-logs to present logs inline
    all_test.go:173: -- test log scope end --
=== RUN   TestAllRegisteredSetup/indexes
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestAllRegisteredSetup_indexes3949903328
    test_log_scope.go:76: use -show-logs to present logs inline
    all_test.go:173: -- test log scope end --
=== RUN   TestAllRegisteredSetup/kv
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestAllRegisteredSetup_kv1210948209
    test_log_scope.go:76: use -show-logs to present logs inline
    all_test.go:173: -- test log scope end --
=== RUN   TestAllRegisteredSetup/querylog
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestAllRegisteredSetup_querylog1104622240
    test_log_scope.go:76: use -show-logs to present logs inline
    all_test.go:173: -- test log scope end --
=== RUN   TestAllRegisteredSetup/schemachange
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestAllRegisteredSetup_schemachange709602276
    test_log_scope.go:76: use -show-logs to present logs inline
    all_test.go:173: -- test log scope end --
=== RUN   TestAllRegisteredSetup/sqlsmith
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestAllRegisteredSetup_sqlsmith450154442
    test_log_scope.go:76: use -show-logs to present logs inline
    all_test.go:173: -- test log scope end --
=== RUN   TestAllRegisteredSetup/tpccmultidb
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestAllRegisteredSetup_tpccmultidb66697029
    test_log_scope.go:76: use -show-logs to present logs inline
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=2</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/test-eng @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAllRegisteredSetup.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-43148

--- a/pkg/ccl/workloadccl/allccl/all_test.go
+++ b/pkg/ccl/workloadccl/allccl/all_test.go
@@ -113,6 +113,7 @@ func TestAllRegisteredImportFixture(t *testing.T) {
 
 func TestAllRegisteredSetup(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	skip.UnderDeadlock(t)
 
 	for _, meta := range workload.Registered() {
 		if bigInitialData(meta) {

